packages/modal/src/Modal.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import { useTransition, animated } from "react-spring";
+import Link from "@crave/farmblocks-link";
+
+import {
+  Wrapper,
+  Overlay,
+  ConstrainedCard,
+  ContentWrapper,
+  Header,
+} from "./Modal.styled";
+
+const Modal = ({
+  isOpen,
+  parentNode,
+  shouldCloseOnOverlayClick,
+  shouldCloseOnEsc,
+  onRequestClose,
+  showCloseButton,
+  children,
+  cardProps,
+  closeProps,
+  className,
+}) => {
+  if (!parentNode) return null;
+
+  const handleKeyDown = event => {
+    if (shouldCloseOnEsc && event.key === "Escape") {
+      onRequestClose?.(event);
```

Comment:
Since the modal can be closed by clicking on the close icon, clicking on overlay and pressing ESC, I wonder if we should add a function that should be called when the modal gets closed `onModalClosed`